skunkworks_crow/src/main/java/org/odk/share/views/ui/settings/SettingsActivity.java

```diff
@@ -162,4 +174,86 @@ private void showPasswordDialog() {
         alertDialog.setCancelable(true);
         alertDialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
     }
+
+    /**
+     * Reset the application settings and the database.
+     */
+    private void resetApplication() {
+        View checkBoxView = View.inflate(this, R.layout.pref_reset_dialog, null);
+        CheckBox cbResetPref = checkBoxView.findViewById(R.id.cb_reset_pref);
+        CheckBox cbResetData = checkBoxView.findViewById(R.id.cb_clear_db);
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        AlertDialog resetDialog = builder.setTitle(getString(R.string.title_reset_settings))
+                .setView(checkBoxView)
+                .setCancelable(false)
+                .setPositiveButton(getString(R.string.ok), (DialogInterface dialog, int which) -> {
+                    progressDialog = new ProgressDialog(this);
+                    progressDialog.setTitle(getString(R.string.resetting));
+                    progressDialog.setMessage(getString(R.string.resetting_msg));
+                    progressDialog.setIndeterminate(true);
+                    progressDialog.setCancelable(false);
+                    progressDialog.show();
+                    dialog.dismiss();
+
+                    if (!cbResetData.isChecked() && !cbResetPref.isChecked()) {
+                        progressDialog.dismiss();
+                        Toast.makeText(this, getString(R.string.reset_select_nothing), Toast.LENGTH_LONG).show();
+                    } else {
+                        startRest(cbResetPref, cbResetData);
+                    }
+                })
+                .setNegativeButton(getString(R.string.cancel), (DialogInterface dialog, int which) -> {
+                    dialog.dismiss();
+                })
+                .create();
+        resetDialog.show();
+    }
+
+    /**
+     * Start resetting the application and presenting the reset result in an {@link AlertDialog}.
+     */
+    private void startRest(CheckBox cbResetPref, CheckBox cbResetData) {
```

Comment:
I think there is a typo in `startRest`, I think this should be changed to `startReset`

Reply:
Fixed